### PR TITLE
doris code

### PR DESCRIPTION
**Pull request by DorisLL**
Add 4 files: 
BR_Doris1115 for Business Rules,
CC_Doris1115 for Computed Columns,
PTables_Doris1115 for Populate Tables, and
SP_Doris1115 for Stored Procedures.

### CODE REVIEW

File: BR_Doris1115.sql [hunk 1]
```diff
@@ -0,0 +1,66 @@
+USE Proj_B1
+
+/*
+4. Enforce the business rule to prevent new inserts into EMP_SHIFT_STATUS for a shift that has already passed
+ */
+CREATE FUNCTION FN_noPassedESS()
```
Comment (andreybutenko) [inline BR_Doris1115.sql]: Put your name in the function so Greg gives you credit :)

File: PTables_Doris1115.sql [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+CREATE PROCEDURE addNewMonth
+@MonthName Varchar(10),
+@MonthNum INT
+AS
+BEGIN TRANSACTION A3
+INSERT INTO tblMONTH (MonthName, MonthNum) VALUES (@MonthName, @MonthNum)
+IF @@error <> 0
+    ROLLBACK TRANSACTION
+ELSE
+    COMMIT TRANSACTION
```
Comment (acohen29) [inline PTables_Doris1115.sql]: Rather then do a procedure, just create the code for a simple insert 5 times for each of the populates. ie. INSERT INTO () VALUES (valueSet1), (valueSet2), etc

File: BR_Doris1115.sql [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+                H.HourName <= (SELECT DATEPART(HOUR, GetDate())))
+        )
+    BEGIN
+        SET @RET = 1
+    END
+RETURN @RET
+END
+
+GO
+ALTER TABLE tblSHIFT
```
Comment (andreybutenko) [inline BR_Doris1115.sql]: You should alter the table `tbl_EMP_SHIFT_STATUS` since that is the transactional table :)

File: BR_Doris1115.sql [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+BEGIN
+    DECLARE @RET INT = 0
+    IF EXISTS (
+        SELECT * FROM tblEMP_SHIFT_STATUS ESS
+            JOIN tblSTATUS St ON St.StatusID = ESS.StatusID
+            JOIN tblSHIFT S ON S.ShiftID = ESS.ShiftID
+            JOIN tblMONTH M ON M.MonthID = S.MonthID
+            JOIN tblDay D ON D.DayID = S.DayID
+            JOIN tblHOUR H ON H.HourID = S.BeginHourID
+        WHERE (St.StatusTitle = 'Worked') OR
```
Comment (andreybutenko) [inline BR_Doris1115.sql]: Is this necessary?

File: BR_Doris1115.sql [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+              (S.[YEAR] < YEAR(GetDate())) OR
+              (S.[YEAR] = YEAR(GetDate() AND M.MonthNum < MONTH(GetDate()))) OR
+              (S.[YEAR] = YEAR(GetDate() AND M.MonthNum = MONTH(GetDate())) AND D.DayName < DAY(GetDate())) OR
+              (S.[YEAR] = YEAR(GetDate() AND M.MonthNum = MONTH(GetDate())) AND D.DayName = DAY(GetDate()) AND
```
Comment (andreybutenko) [inline BR_Doris1115.sql]: `DateUpdated` cannot be past the Shift's time

Comment (andreybutenko) [reply BR_Doris1115.sql]: Alex suggests changing `GetDate()` to `DateUpdated`

**Review by andreybutenko — CHANGES_REQUESTED** [review]
Great work!

File: SP_Doris1115.sql [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+    SET @EmpID = (SELECT SCOPE_IDENTITY())
+
+    INSERT INTO tblEMPLOYEE_POSITION(PositionID, EmployeeID, BeginDate, EndDate)
+    VALUES (@PosID, @EmpID, @BeginDate, @EndDate)
+
+IF @@error <> 0
+    ROLLBACK TRANSACTION
+ELSE
+    COMMIT TRANSACTION
+
```
Comment (jelauria) [inline SP_Doris1115.sql]: Population of table not required; thanks for writing them out though! You can save these execute statements into a separate file (do this for the second sproc as well) to make the file shorter (and more readable). Other than that, no errors found; good job!

**Review by jelauria — CHANGES_REQUESTED** [review]

File: BR_Doris1115.sql (hunk 4 above)
Comment (DorisLL) [reply BR_Doris1115.sql]: What if there is a shift that has not been worked but the time passed? Is it possible?

File: BR_Doris1115.sql [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+4. Enforce the business rule to prevent new inserts into EMP_SHIFT_STATUS for a shift that has already passed
+ */
+CREATE FUNCTION dorisFN_noPassedESS()
+RETURNS INT
+AS
+BEGIN
+    DECLARE @RET INT = 0
+    IF EXISTS (
+        SELECT * FROM tblEMP_SHIFT_STATUS ESS
+            JOIN tblSTATUS St ON St.StatusID = ESS.StatusID
```
Comment (acohen29) [inline BR_Doris1115.sql:12]: Needs to connect to tblShift

**Pull request merged**